app/gui/src/project-view/components/visualizations/TableVisualization.vue

```diff
@@ -219,7 +223,7 @@ function formatText(params: ICellRendererParams) {
       })
 
   const replaceLinks = replaceSpaces.replace(
-    /https?:\/\/([-()_.!~*';/?:@&=+$,A-Za-z0-9])+/g,
+    /https?:\/\/([a-zA-Z0-9-]+(\.[a-zA-Z0-9-]+)+)(:[0-9]+)?(\/[-()_.!~*';/?:%@&=+$,A-Za-z0-9]*)?/,
     (url: string) => `<a href="${url}" target="_blank" class="link">${url}</a>`,
   )
```

Comment:
This is complex enough to warrant extraction to a function rather than duplicating it.